test/integration/thesis_test.rb

```diff
@@ -167,6 +167,57 @@ def teardown
     assert_equal new_thesis_path, path
   end
 
+  test 'a confirmation email is sent on form submit' do
+    user = users(:basic)
+    thesis = theses(:two)
+
+    mock_auth(user)
+
+    ClimateControl.modify DISABLE_ALL_EMAIL: 'false' do
+      email = ReceiptMailer.receipt_email(thesis, user)
+
+      assert_emails 1 do
+        patch thesis_path(thesis), params: { thesis: thesis.serializable_hash }
+      end
+      assert_equal ['basic@example.com'], email.to
+      assert_equal 'Your thesis information submission', email.subject
+    end
+  end
+
+  test 'a confirmation email is not sent when emails are disabled' do
+    user = users(:basic)
+    thesis = theses(:two)
+
+    mock_auth(user)
+
+    ClimateControl.modify DISABLE_ALL_EMAIL: 'true' do
+      email = ReceiptMailer.receipt_email(thesis, user)
+
+      assert_emails 0 do
+        patch thesis_path(thesis), params: { thesis: thesis.serializable_hash }
+      end
+    end
+  end
+
+  test 'confirmation emails are sent to current user, not the author' do
+    author = users(:basic)
+    user = users(:thesis_admin)
+    thesis = theses(:two)
+
+    mock_auth(users(:thesis_admin))
+
+    ClimateControl.modify DISABLE_ALL_EMAIL: 'false' do
+      email = ReceiptMailer.receipt_email(thesis, user)
+
+      assert_emails 1 do
+        patch thesis_path(thesis), params: { thesis: thesis.serializable_hash }
+      end
+      assert_not_equal author.email, email.to
+      assert_equal ['thesis_admin@example.com'], email.to
+      assert_equal 'Your thesis information submission', email.subject
+    end
+  end
+
```

Comment:
Do we need tests to confirm whether an email is sent from a new thesis form? At the moment the `create` method ends up calling `update`, so the email gets sent that way - but if that logic ever changes, we'll lose the email.

Reply:
That's a good question. I think it makes sense as a kindness to our future selves. 👍